### PR TITLE
Samples: Bluetooth: Fix PAST support for broadcast sink

### LINKED ISSUE #68212: LE Audio: Broadcast audio sink sample not handling PAST correctly
**Describe the bug**
If both sides support PAST, then the sample should, but isn't, setting the PA state to `BT_BAP_PA_STATE_INFO_REQ`.

**To Reproduce**
Connect the sample with a broadcast assistant that supports PAST, and see that the PAST timer expires. 

**Expected behavior**
In the `pa_sync_req_cb` function we call `pa_sync_past`. 

The function is effectively missing the following code:
```c
		if (err != 0) {
			err = bt_bap_scan_delegator_set_pa_state(src_id,
								 BT_BAP_PA_STATE_INFO_REQ);
			if (err != 0) {
				return err;
			}

		}
```

**Impact**
Minor as we can disable PAST support by setting `CONFIG_BT_PER_ADV_SYNC_TRANSFER_RECEIVER=n`. 

**Logs 
```
*** Booting nRF Connect SDK zephyr-v3.5.0-3023-g37c7cef103cb ***
[00:00:00.280,761] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)
[00:00:00.280,792] <inf> bt_hci_core: HW Variant: nRF53x (0x0003)
[00:00:00.280,822] <inf> bt_hci_core: Firmware: Standard Bluetooth controller (0x00) Version 112.313 Build 1086269351
[00:00:00.282,562] <inf> bt_hci_core: Identity: EE:27:7B:87:FA:78 (random)
[00:00:00.282,592] <inf> bt_hci_core: HCI: version 5.4 (0x0d) revision 0x2166, manufacturer 0x0059
[00:00:00.282,623] <inf> bt_hci_core: LMP: version 5.4 (0x0d) subver 0x2166
[00:00:00.282,623] <inf> main: Bluetooth initialized

[00:00:00.282,653] <inf> main: Starting advertising

[00:00:00.284,423] <inf> main: Waiting for Broadcast Assistant

[00:02:02.313,201] <inf> main: Connected: 55:37:D0:B0:72:BA (random)
[00:02:05.151,611] <dbg> bt_bap_scan_delegator: read_recv_state: Index 0: Not active
[00:02:05.241,760] <dbg> bt_bap_scan_delegator: recv_state_cfg_changed: value 0x0001
[00:02:38.632,904] <wrn> bt_att: Unhandled ATT code 0x1d
[00:02:47.753,326] <dbg> bt_bap_scan_delegator: write_control_point: Data
                                                01                                               |.
[00:02:47.753,356] <dbg> bt_bap_scan_delegator: write_control_point: Assistant starting scanning
[00:02:48.953,857] <dbg> bt_bap_scan_delegator: write_control_point: Data
                                                02 00 17 ff 50 f4 38 3c  01 74 80 72 01 68 01 01 |....P.8< .t.r.h..
                                                ff ff ff ff 07 03 02 04  00 02 05 01             |........ ....
[00:02:48.953,887] <dbg> bt_bap_scan_delegator: write_control_point: Assistant adding source
[00:02:48.953,918] <inf> main: pa_sync_req_cb
[00:02:48.953,948] <inf> main: pa_sync_past
[00:02:48.954,345] <inf> main: Syncing with PAST: 0
[00:02:48.954,376] <dbg> bt_bap_scan_delegator: scan_delegator_add_source: Index 0: New source added: ID 0x00
[00:02:48.954,559] <dbg> bt_bap_scan_delegator: bt_debug_dump_recv_state: Receive State[0]: src ID 0, addr 3C:38:F4:50:FF:17 (public), adv_sid 1, broadcast_id 0x728074, pa_sync_state 0, encrypt state 0, num_subgroups 1
[00:02:48.954,620] <dbg> bt_bap_scan_delegator: bt_debug_dump_recv_state:       Subgroup[0]: BIS sync 0 (requested 4294967295), metadata_len 7, metadata: 03020400020501
[00:02:48.954,742] <inf> main: recv_state_updated_cb
[00:02:48.954,742] <inf> main: bis_sync_req_cb
[00:02:48.954,772] <inf> main: BIS sync request received for 0x20004454: 0xffffffff
[00:02:48.954,772] <inf> main: source id = 0, adv_id = 1, broadcast id = 728074
[00:02:48.954,986] <inf> main: Waiting for PA synced

[00:02:59.754,608] <dbg> bt_bap_scan_delegator: bt_debug_dump_recv_state: Receive State[0]: src ID 0, addr 3C:38:F4:50:FF:17 (public), adv_sid 1, broadcast_id 0x728074, pa_sync_state 3, encrypt state 0, num_subgroups 1
[00:02:59.754,699] <dbg> bt_bap_scan_delegator: bt_debug_dump_recv_state:       Subgroup[0]: BIS sync 0 (requested 4294967295), metadata_len 7, metadata: 03020400020501
[00:02:59.754,791] <inf> main: recv_state_updated_cb
[00:02:59.754,791] <inf> main: PA timeout
```

**Environment (please complete the following information):**
 - Commit SHA or Version used: 682f38fa3663e483dd381d868b1e6f9eb6f4d2fb

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
Fix missing support for PAST in the broadcast audio sink sample, where it previously failed to request PAST from the broadcast assistant by not setting the PA state to BT_BAP_PA_STATE_INFO_REQ.

fixes https://github.com/zephyrproject-rtos/zephyr/issues/68212

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: samples/bluetooth/broadcast_audio_sink/src/main.c [hunk 1]
```diff
@@ -667,16 +666,29 @@ static int pa_sync_req_cb(struct bt_conn *conn,
...
+
+			return err;
+		}
+
 		k_sem_give(&sem_past_request);
-	} else {
-		/* start scan */
-		err = 0;
+
+		err = bt_bap_scan_delegator_set_pa_state(recv_state->src_id,
```
Comment (pin-zephyr) [inline samples/bluetooth/broadcast_audio_sink/src/main.c:680]: should we set the PA state to BT_BAP_PA_STATE_INFO_REQ to request PAST prior to calling pa_sync_past() in the line 671 ?

**Review by pin-zephyr — CHANGES_REQUESTED** [review]

File: samples/bluetooth/broadcast_audio_sink/src/main.c (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:680]: > should we set the PA state to BT_BAP_PA_STATE_INFO_REQ to request PAST prior to calling pa_sync_past() in the line 671 ?

Hmm, good question, but I would argue no. If we (for whatever reason) fail to subscribe to PAST (which is what `pa_sync_past` does), then we don't need to update the state. I think we makes sense to update the state, when we reach a point that we are waiting for the remote device

Comment (pin-zephyr) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:680]: how about if we are not allowed to set the state to be BT_BAP_PA_STATE_INFO_REQ for some reason (e.g Could not find recv_state by src_id ....), do we still need to subscribe to PAST by pa_sync_past() ?

**Review by pin-zephyr — CHANGES_REQUESTED** [review]

File: samples/bluetooth/broadcast_audio_sink/src/main.c (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:680]: Hmm, that is a fair point. 

So basically we have the option to do

1. Subscribe to past
2. Set state (if this fails, no state change was done, and we can return error)

or

1. Set state (if this fails, no state change was done, and we can return error)
2. Subscribe to past
3. If 2. fails, we need to restore state (and what if this fails?)

I would be OK with changing it to the latter, but it is a more complex solution, where as subscription to PAST is not really hurtful (and cannot be undone), even if we don't receive PAST from the peer. 

Let me know if you want me to change it, but my immediate opinion would be to keep as is to keep it simple

**Review by kruithofa — APPROVED** [review]

File: samples/bluetooth/broadcast_audio_sink/src/main.c (hunk 1 above)
Comment (pin-zephyr) [reply samples/bluetooth/broadcast_audio_sink/src/main.c:680]: I see, it is ok for me to choose option 1 :-)

**Review by pin-zephyr — APPROVED** [review]
LGTM

**Pull request merged**